vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/scatter_plotly.tsx

```diff
@@ -0,0 +1,230 @@
+    if (hide.includes(keys[ind])) delete values[keys[ind]];
+  }
+  return values;
+};
+
+let showAdvanced = false;
+
+export default function ScatterPlotly({
+  data, onChange, ...props
+}: WithInputParams<{}, DataEnvelope<any>, any>) {
+  const hide = useMemo(() => data && data['hide'], [data]);
+  const defaultValue = useMemo(() => remove_hidden(defaults, hide), [hide]);
+  const value = useSetsDefault(defaultValue, props.value, onChange);
+
+  const options: DataEnvelope<string> = useMemo(() => {
+    if (data == null) return {};
+
+    return data['data_options'] || nulled_vals(data['data_frame']);
+  }, [data]);
+
+  const updateValue = (newValue: any, key: string, prevValues = value) => {
+    prevValues[key] = newValue;
+    onChange(some(prevValues));
+  };
+
+  // Component Setups
+  const string_input = (
+    key: string = "filler", value: string | number | boolean = "filler",
+    label: string = 'hello') => {
+      return (
+        <TextInput
+          key={key}
+          header={label}
+          value={value}
+          onChange={(newValue: string) => updateValue(newValue, key)}
+        />
+      )
+    };
+
+  const nestable_dropdown_input = (
+    key: string = "filler", value: string | null,
+    label: string, options: DataEnvelope<null>) => {
+      
+      return(
+        <div key={key}>
+          {label}
+          <NestedSelectAutocompleteInput
+            data={{options}}
+            value={maybeOfNullable(value)}
+            onChange={(val) => updateValue(withDefault(val, null), key)}
+          />
+        </div>
+      )
+    }
+  
+  const dropdown_input = (
+    key: string = "filler", value: string | null,
+    label: string, defaultValue: string | null = null, options: string[]) => {
+      
+      return(
+        <div key={key}>
+          {label}
+          <DropdownAutocomplete
+            list={options}
+            defaultValue={defaultValue}
+            value={value}
+            onChange={(val: string) => updateValue(val, key)}
+          />
+        </div>
+      )
+    }
+
+  const checkbox_input = (
+    key: string = "filler", value: boolean = false,
+    label: string) => {
+
+      return(
+          <div key={key}>
+            {label}
+            <input id={key} type='checkbox'
+                  checked={value}
+                  onChange={() => updateValue(!value, key)} />
+          </div>
+      )
+    }
+
+  const slider_input = (
+    key: string = "filler", value: number,
+    label: string, min: number = 0.1, max: number = 20) => {
+
+      return(
+          <div key={key}>
+            {label}
+            <Slider
+              value={value}
+              onChange={(event, newValue) => updateValue(newValue as number, key)}
+              min={min}
+              max={max}
+              valueLabelDisplay="auto"
+            />
+          </div>
+      )
+    }
+
+  const extra_inputs: DataEnvelope<any[]> = useMemo(() => {
+    return {
```

Comment:
Are some of these values "default" values for the inputs? It seems inconsistent to include these in the UI component, when most other default values come from the server...will that be confusing to someone making a new workflow?

Reply:
Text labels and default values yes, but for the bits behind the `advanced options` button.  Initial values, but I don't think a workflow editor should ever need to concern themself with them... none are workflow context dependent, so easier to create them within the tsx than to require the workflow author to make a `defaults` input.

This is part of the ugliness of way I have the "The structure of this component is as a nested set of other configurable inputs." set up currently.

Reply:
I get that, and I think it's a reasonable approach.  Certainly, _assuming reasonable complexity in the component to make workflow creation easier_ is a worthy goal.  I think we can reduce the complexity of this design together, let's pair early next week on it.